### DISCUSSION TITLE
Add a hook that prompts the user to SIWE.

### DISCUSSION THREAD

**Discussion by thevolcanomanishere** [rainbow-me/rainbowkit#1937] [Ideas]
If an application uses `<RainbowKitAuthenticationProvider/>`, the user is prompted to sign a message immediately to sign-in. There are many use cases where I want to connect a users wallet, and only request a signature when there is some authentication needed in order to see something privileged or write to my backend. Right now, you can either connect a users wallet and then do all the SIWE stuff manually OR use `<RainbowKitAuthenticationProvider/>` and you are forced to have the user sign a message immediately.

Allow us to disable the automatic verification popup.

```javascript
<RainbowKitAuthenticationProvider
verifyImmediately={false}
>
```

A simple hook like the following would be very useful and enable more user friendly use cases.

```javascript
const { authenticateWithSiwe } = useAccount();

const onClickButton = () => {
    authenticateWithSiwe() // The "Verify your account" modal pops up now.
}
```

**Comment by thevolcanomanishere** [comment]
So after inspecting the `RainbowKitAuthenticationProvider` code I found the `enabled` setting! I can use that to stop the Verify request after a user connects, but I would still need the `authenticateWithSiwe` hook to programatically pop-up the verify message when a user intends to do something that requires authentication. Right now, if I flip `enabled` to true, the user needs to click the wallet button to bring up the prompt. 

Currently doing this which is not ideal

```typescript
    setTimeout(() => {
      // Find the button with the specific data-testid
      const button = document.querySelector(
        '[data-testid="rk-connect-button"]'
      ) as HTMLElement;

      // If the button exists, simulate a click event
      if (button) {
        button.click();
      }
    }, 1000);
```

**Reply by 0xmkh** [reply]
You don't have to do this. You can just use `openConnectModal` from `useConnectModal` hook whenever you want to open the SIWE modal, but remember user always has to connect his wallet first to perform a signature request.

Like this:

```
const { openConnectModal } = useConnectModal();

<button onClick={() => {
    if (openConnectModal) {
      openConnectModal();
    }
  }}
>
  open modal
</button>
```

**Comment by 0xmkh** [comment]
@thevolcanomanishere Can you try passing `status` prop to be `'authenticated'` after user connects his wallet right away and whenever your backend requests a change in auth you can switch immediately to  `'unauthenticated'` ? [Here](https://www.rainbowkit.com/docs/custom-authentication#providing-the-authentication-state) is the docs for the `RainbowKitAuthenticationProvider`.

Can you give more scenarios / examples of what you're trying to achieve ? As far as i understood you don't want users to get a SIWE modal popup right away, but whenever you wish to right ?

**Reply by thevolcanomanishere** [reply]
I am using the status + enabled prop to control when the sign request comes up. I am also watching for the button to appear to automatically click it for the user. That is all working well, but the auto button clicking stuff is a hack due to the fact I cant control that automatically skipping the user interaction step that is required to get the signature prompt on their wallet.

My scenario here is using rainbow kit in a point of sale system. The user should only have to scan the QR code for either walletconnect or coinbase wallet, and the system should be able to programmatically control if a signature is requested or not. I want to remove the extra user action step in between to make this slicker. A simple `siweMessagePopupEnabled` flag would do the trick.

Also being able to tell Rainbow kit to always display the QR code regardless of the environment it is in. That way I can determine if RainbowKit shows the QR code for WalletConnect and Coinbase instead of the Rainbow figuring that out for me. If I use RainbowKit on an iPad for instance, when I click the coinbase wallet icon, nothing happens (even with Coinbase Wallet installed). I should be able to force RainbowKit to always display the QR code UNLESS I decide its appropriate to try and connect to some local wallet instance like Metamask etc.

**Reply by 0xmkh** [reply]
Sorry just trying to understand your approach here. Can't you just switch `status` prop to use `'authenticated'` value if you want to display the QR code instead of requiring the signature popup ? Since if it's `'authenticated'` you'll not be forced to sign the message.
